silx/image/sift/match.py

```diff
@@ -0,0 +1,335 @@
+#!/usr/bin/env python
+# -*- coding: utf-8 -*-
+#
+#
+#    Project: Sift implementation in Python + OpenCL
+#             https://github.com/silx-kit/silx
+#
+# Permission is hereby granted, free of charge, to any person
+# obtaining a copy of this software and associated documentation
+# files (the "Software"), to deal in the Software without
+# restriction, including without limitation the rights to use,
+# copy, modify, merge, publish, distribute, sublicense, and/or sell
+# copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following
+# conditions:
+#
+# The above copyright notice and this permission notice shall be
+# included in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND,
+# EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES
+# OF MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+# NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT
+# HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY,
+# WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR
+# OTHER DEALINGS IN THE SOFTWARE.
+
+"""
+Contains a class for creating a matching plan, allocating arrays, compiling kernels and other things like that
+"""
+
+from __future__ import division, print_function, with_statement
+
+__authors__ = ["Jérôme Kieffer"]
+__contact__ = "jerome.kieffer@esrf.eu"
+__license__ = "MIT"
+__copyright__ = "European Synchrotron Radiation Facility, Grenoble, France"
+__date__ = "25/08/2016"
+__status__ = "beta"
+import os
+import logging
+import sys
+import threading
+import gc
+import numpy
+from .param import par
+from silx.opencl import ocl, pyopencl
+from .utils import calc_size, get_opencl_code
+logger = logging.getLogger("sift.match")
+if not pyopencl:
+    logger.warning("No PyOpenCL, no sift")
+
+
+class MatchPlan(object):
+    """Plan to compare sets of SIFT keypoint
+
+    siftp = sift.MatchPlan(devicetype="GPU")
+    kp = siftp.match(kp1,kp2)
```

Comment:
Code formatting issue.
In the html generated by sphinx, the 2 lines of code are displayed on the same line. 
It should be:
```
Plan to compare sets of SIFT keypoint

.. code-block:: python

    siftp = sift.MatchPlan(devicetype="GPU")
    kp = siftp.match(kp1,kp2)
```